### PR TITLE
PanelChrome: Make loading animation speed the same across panels 

### LINKED ISSUE #60672: PanelChrome: Loading bar should be relatively same size and speed on all different panels

### PR DESCRIPTION
**Pull request by torkelo**
* [x] Fixes useStyles2 use so that memoization works, moving the dynamic bit to just CSSProperties & style prop
* [x] Makes animation speed variable by width so that the animation moves at the same speed across different panels widths
* [x] Makes animation border height 1px instead of 2 (happy for feedback on that, can change back) 
  
Fixes https://github.com/grafana/grafana/issues/60672

### CODE REVIEW

File: packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx [hunk 1]
```diff
@@ -1,48 +1,45 @@
...
-import React from 'react';
-
-import { GrafanaTheme2 } from '@grafana/data';
+import React, { CSSProperties } from 'react';
 
 import { useStyles2 } from '../../themes';
 
 export interface LoadingBarProps {
-  width?: string;
-  height?: string;
```
Comment (academo) [inline packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:10]: Is removing this height prop intended? Levitate reports a breakage in the API by removing it but I could not find any plugin using it at the moment.

I see that removing this prop removes some flexibility from the component and set a static height value. Could we leave this height customizable?

Comment (torkelo) [reply packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:10]: @academo yes, but this is an internal component that not exposed anywhere

Comment (torkelo) [reply packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:10]: at least I thought it was an internal implementation detail of PanelChrome. But looks like a standalone component, not sure why we did that given it's such an early state component that does not need to be exported initially. I think we should remove it from being exported until we finalize the design of it and see more use cases for it

File: packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx [hunk 2]
```diff
@@ -1,48 +1,44 @@
...
   ariaLabel?: string;
 }
 
-export function LoadingBar({ width, height, ariaLabel = 'Loading bar' }: LoadingBarProps) {
-  const styles = useStyles2(getStyles(width, height));
+const MILLISECONDS_PER_PIXEL = 2.4;
+
+export function LoadingBar({ width, ariaLabel = 'Loading bar' }: LoadingBarProps) {
+  const styles = useStyles2(getStyles);
+  const animationSpeed = Math.min(Math.max(Math.round(width * MILLISECONDS_PER_PIXEL), 500), 4000);
```
Comment (leventebalogh) [inline packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:15]: Nit: I think by extracting the min and max timing values to constants as well we could make the code easier to read.

```javascript
const MILLISECONDS_PER_PIXEL = 2.4;
const MIN_SPEED = 500; // ms
const MAX_SPEED = 4000; // ms
```

```suggestion
  const animationSpeed = Math.min(Math.max(Math.round(width * MILLISECONDS_PER_PIXEL), MIN_SPEED), MAX_SPEED);
```

Comment (torkelo) [reply packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx]: done!

**Comment by torkelo** [comment]
@polibb @axelavargas Patrick gave the :+1:  to this, can you take a look if the code make sense?

File: packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx [hunk 3]
```diff
@@ -1,48 +1,46 @@
...
+        transform: 'translateX(-50%)',
+      },
+      '100%': {
+        transform: `translateX(100%)`,
+      },
     }),
     bar: css({
-      width: barWidth,
-      height: loadingHeigth,
+      width: '28%',
```
Comment (axelavargas) [inline packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:42]: 🤔 I noticed we are not using the width we are passing down, we are only using it to calculate the durationMs, is it not needed here?

Comment (torkelo) [reply packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:42]: I dont think so, we where not really using before either. Was just a hard coded percentage of width. Which I think should be the same in all places we use this to be consistent

**Review by leventebalogh — APPROVED** [review]
LGTM 👍

File: packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx (hunk 3 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:42]: @axelavargas the width passed down is for the full width where the animation takes place, this is the width of the colored gradient moving along. and that can just be a fixed percentage (or absolute width)

Comment (axelavargas) [reply packages/grafana-ui/src/components/LoadingBar/LoadingBar.tsx:42]: Aha! Now I get it 🙂

**Review by axelavargas — APPROVED** [review]

**Pull request merged**